pkg/cmd/codespace/codespace_selector.go

```diff
@@ -0,0 +1,138 @@
+package codespace
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/v2/internal/codespaces/api"
+	"github.com/spf13/cobra"
+)
+
+type CodespaceSelector interface {
+	// Select fetches a codespace from the API according to the criteria set at its creation, prompting the user if needed
+	Select(ctx context.Context) (codespace *api.Codespace, err error)
+
+	// SelectName fetches a codespace name in the same way as Select, but may avoid API calls
+	SelectName(ctx context.Context) (name string, err error)
+}
+
+type codespaceSelector struct {
+	api apiClient
+
+	repoName      string
+	codespaceName string
+}
+
+var errNoFilteredCodespaces = errors.New("you have no codespaces meeting the filter criteria")
+
+// AddCodespaceSelector add codespace filtering parameters to the give command and returns a CodespaceSelector which applies them
+func AddCodespaceSelector(cmd *cobra.Command, app *App) *codespaceSelector {
+	cs := &codespaceSelector{api: app.apiClient}
+
+	cmd.Flags().StringVarP(&cs.codespaceName, "codespace", "c", "", "Name of the codespace")
+	cmd.Flags().StringVarP(&cs.repoName, "repo", "R", "", "Filter codespace selection by repository name (user/repo)")
+
+	cmd.MarkFlagsMutuallyExclusive("codespace", "repo")
+
+	return cs
+}
+
+// AddDeprecatedRepoShorthand adds a -r parameter (deprecated shorthand for --repo)
+// which instructs the user to use -R instead.
+func (cs *codespaceSelector) AddDeprecatedRepoShorthand(cmd *cobra.Command) error {
```

Comment:
This method at least needs the selector as a param if not be a method on the selector. It is used in `delete` now, but will be on a couple more once I finish the implementation. It needs the selector so that it can also set the `repoName`

Reply:
Ah I misread this, and missed the `cs` reference.